include/zephyr/bluetooth/conn.h

```diff
@@ -1173,8 +1173,11 @@ struct bt_conn_cb {
 	 * @ref bt_conn_unref which is used by the BT stack. Making
 	 * Bluetooth API calls in this context is error-prone and strongly
 	 * discouraged.
+	 *
+	 * @param conn Connection object that was returned. Can ONLY be used
+	 *             for comparison purposes, not API calls.
```

Comment:
I take issue with this description. Freed pointers are not strictly safe for comparison, in the usual sense of object identity equality. There is no guarantee that the memory has not been repurposed for a new object.

We should rephrase this. The purpose of this new pointer value argument can only be that it identifies a "connection slot" which at any point in time can hold at most one connection object.

To refer to a connection slot, I would be more inclined to give the `bt_conn_id(conn)` number instead of the pointer value. What do you think of the practicalities of this?

Reply:
I have no objections if this is the preferred method. All I require is a way to know when the `conn` object provided to the `bt_le_ext_adv_cb.connected` callback is free again.